src/makeCalculator.js

```diff
@@ -37,7 +37,49 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  const add = (number) => {
+    result += number;
+  };
+
+  const subtract = (number) => {
+    result -= number;
+  };
+
+  const multiply = (number) => {
+    result *= number;
+  };
+
+  const divide = (number) => {
+    result /= number;
+  };
+
+  const reset = () => {
+    result = 0;
+
+    return calc;
+  };
+
+  const operate = (operation, number) => {
+    operation(number);
+
+    return calc;
+  };
```

Comment:
What if I make 2 objects with makeCalculator() and pass the method of second one to the `operate` of first one?
![image](https://github.com/mate-academy/js_advanced_calculator/assets/46566190/edd9e3ec-7a69-4a6e-8f03-b0be7e7d26e1)

I think that it is suppose to change the object on which `operate` was called. Think of using methods that are changing the context  of the invoking function.

Reply:
@dinaminahi honestly, I couldn't understand how can I do it else, only like rewrite solution and put all that in 1 common object. Now it works, if I based on your example